backend/src/main/java/gov/cdc/usds/simplereport/api/model/filerow/TestResultRow.java

```diff
@@ -484,6 +513,29 @@ private boolean validModelTestPerformedCombination(
                     removeTrailingAsterisk(equipmentModelName), testPerformedCode));
   }
 
+  private boolean validResultsOnlyIncludeActiveDiseases(
+      String equipmentModelName, String testPerformedCode) {
+
+    Map<String, DeviceType> referenceList =
+        resultsUploaderCachingService.getModelAndTestPerformedCodeToDeviceMap();
+    String deviceTypeToCheckKey =
+        ResultsUploaderCachingService.getKey(
+            removeTrailingAsterisk(equipmentModelName), testPerformedCode);
+
+    DeviceType deviceTypeToCheck = referenceList.get(deviceTypeToCheckKey);
+    List<String> supportedDiseaseNamesToCheck =
+        deviceTypeToCheck.getSupportedDiseaseTestPerformed().stream()
+            .map(DeviceTypeDisease::getSupportedDisease)
+            .map(SupportedDisease::getName)
+            .toList();
+
+    if (!featureFlagsConfig.isHivBulkUploadEnabled()
+        && supportedDiseaseNamesToCheck.contains("HIV")) {
+      return false;
+    }
+    return true;
```

Comment:
intellij says that I can simplify this if / else to

```
return featureFlagsConfig.isHivBulkUploadEnabled()
              || !supportedDiseaseNamesToCheck.contains("HIV");
```
which is logically equivalent but is much less readable in my opinion. Leaving this in this form unless someone heavily disagrees